Modules/_collectionsmodule.c

```diff
@@ -243,37 +213,20 @@ Remove and return the rightmost element.
 
 static PyObject *
 deque_pop_impl(dequeobject *deque)
-/*[clinic end generated code: output=2e5f7890c4251f07 input=55c5b6a8ad51d72f]*/
+/*[clinic end generated code: output=21dbf03cd7259ec9 input=4b3754030e7a5478]*/
 {
+    // TODO(Matthias): do we need to lock anything?
     PyObject *item;
-    block *prevblock;
-
     if (Py_SIZE(deque) == 0) {
         PyErr_SetString(PyExc_IndexError, "pop from an empty deque");
         return NULL;
     }
-    item = deque->rightblock->data[deque->rightindex];
-    deque->rightindex--;
+    item = deque->ob_item[(deque->first_element + Py_SIZE(deque) - 1) & (deque->allocated - 1)];
     Py_SET_SIZE(deque, Py_SIZE(deque) - 1);
+    deque_maybe_shrink(deque);
     deque->state++;
-
-    if (deque->rightindex < 0) {
-        if (Py_SIZE(deque)) {
-            prevblock = deque->rightblock->leftlink;
-            assert(deque->leftblock != deque->rightblock);
-            freeblock(deque, deque->rightblock);
-            CHECK_NOT_END(prevblock);
-            MARK_END(prevblock->rightlink);
-            deque->rightblock = prevblock;
-            deque->rightindex = BLOCKLEN - 1;
-        } else {
-            assert(deque->leftblock == deque->rightblock);
-            assert(deque->leftindex == deque->rightindex+1);
-            /* re-center instead of freeing a block */
-            deque->leftindex = CENTER + 1;
-            deque->rightindex = CENTER;
-        }
-    }
+    // TODO(Matthias): consider shrinking the deque, if we are below some
```

Comment:
Only shrink on inserts.  Same justification as for dicts.  Write down a comment explaining this.